app/src/main/java/com/raizlabs/android/databasecomparison/MainActivity.java

```diff
@@ -1,58 +1,213 @@
+        chartEntrySets.put(SprinklesTester.FRAMEWORK_NAME, new ArrayList<BarEntry>());
+    }
+    private int getFrameworkColor(String framework) {
+        // using the 300 line colors from http://www.google.com/design/spec/style/color.html#color-color-palette
+        switch (framework) {
+            case DBFlowTester.FRAMEWORK_NAME:
+                return Color.rgb(0xE5,0x73,0x73); // red
+            case AATester.FRAMEWORK_NAME:
+                return Color.rgb(0xF0, 0x62, 0x92); // pink
+            case GreenDaoTester.FRAMEWORK_NAME:
+                return Color.rgb(0xBA, 0x68, 0xC8); // purple
+            case OrmLiteTester.FRAMEWORK_NAME:
+                return Color.rgb(0x4D, 0xB6, 0xAC); // teal
+            case SprinklesTester.FRAMEWORK_NAME:
+                return Color.rgb(0x79, 0x86, 0xCB); // indigo
+            case SugarTester.FRAMEWORK_NAME:
+                return Color.rgb(0x64, 0xB5, 0XF6); // blue
+            default:
+                return Color.WHITE;
+        }
+    }
+    private void addChartData(String framework, String category, long value) {
+        BarEntry entry = new BarEntry(value, category.equals(SAVE_TIME) ? 0 : 1);
+        chartEntrySets.get(framework).add(entry);
+    }
 
+    private void enableButtons(boolean enabled) {
+        simpleTrialButton.setEnabled(enabled);
+        complexTrialButton.setEnabled(enabled);
     }
 
-    public static void logTime(long startTime, String name) {
-        Log.e(MainActivity.class.getSimpleName(), name + " took: " + (System.currentTimeMillis() - startTime));
+    /**
+     * runs simple benchmarks (onClick from R.id.simple)
+     * @param v button view
+     */
+    public void runSimpleTrial(View v) {
+        setBusyUI(true, getResources().getString(R.string.simple));
+        resetChart();
+        final MainActivity mainActivity = this;
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                OrmLiteTester.testAddressItems(mainActivity);
+                GreenDaoTester.testAddressItems(mainActivity);
+                DBFlowTester.testAddressItems(mainActivity);
+                SprinklesTester.testAddressItems(mainActivity);
+                AATester.testAddressItems(mainActivity);
+                SugarTester.testAddressItems(mainActivity);
+                mainActivity.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        mainActivity.setBusyUI(false, null);
+                        initChart("Simple");
+                    }
+                });
+            }
+        }).start();
+    }
+
+    /**
+     * runs complex benchmarks (onClick from R.id.complex)
+     * @param v button view
+     */
+    public void runComplexTrial(View v) {
+        setBusyUI(true, getResources().getString(R.string.complex));
+        resetChart();
+        final MainActivity mainActivity = this;
```

Comment:
#req may leak activity here. I think we should utilize some kind of `AsyncTask` outside of this activity that contains a callback when the result finishes and the `Activity` can register a listener on it.